src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-realm-form/rgw-multisite-realm-form.component.ts

```diff
@@ -37,11 +32,17 @@ export class RgwMultisiteRealmFormComponent implements OnInit {
     public actionLabels: ActionLabelsI18n,
     public rgwRealmService: RgwRealmService,
     public notificationService: NotificationService,
-    public docService: DocService
+    public docService: DocService,
+    @Optional() @Inject('action') public action: string,
+    @Optional() @Inject('resource') public resource: string,
+    @Optional() @Inject('info') public info: any,
+    @Optional() @Inject('multisiteInfo') public multisiteInfo: object[],
+    @Optional() @Inject('defaultsInfo') public defaultsInfo: string[],
+    @Optional() @Inject('editing') public editing: boolean
```

Comment:
I used Inject mostly because its what i saw in the examples and mocks provided in carbon. I didnt look into Injectable though so I cant say more here..

Reply:
@dnyanee1997 agree we need to be consistent, but these are smaller steps to achieve that.

We do use @Injectable and that is more than @Inject , so obviously its not 100% achieved but we can keep moving. Also this is what angular dev style suggests hence we must follow that.

Thanks

Reply:
I digged into that , and saw that for modal we are using inject only.
We can improve on that for sure, not blocking this PR.